feat: Add support for onFocus and filterPattern in Input Component

RajuSudhar opened the pull request:
- Add rightImageUrl property to InputProperties for displaying icon on right side
- Add filterPattern property to allow custom character filtering beyond default numeric filtering
- Enhance Input component structure with right content slot and image support
- Add comprehensive CSS variables for right content and image styling including hover states
- Update InputButton component styling to remove default input margins and shadows
- Improve text processing logic with configurable filtering patterns

**Dev Proof**
[Video 1](https://drive.google.com/file/d/1NbPROkNNfpYKzVRuMhrdh9GlsMl8ToUx/view?usp=sharing)
[Video 2](https://drive.google.com/file/d/1DGzNVEaN90KijdU5hce3CPHcdKZCYPfs/view?usp=sharing)
[Video 3 - Slots and Image with reordering support ](https://drive.google.com/file/d/1QYbKOAYq1IRG8Lx9_bDEJ8URN67-dW55/view?usp=drive_link)
[Video 4 - Proper backward compatibility](https://drive.google.com/file/d/1lJB-mezG-3pRVFsIhN01my0Qz_T4Ds4v/view?usp=drive_link)

sandeepsj commented on src/lib/Input/Input.svelte:
```diff
@@ -206,7 +242,7 @@
   input {
     box-sizing: var(--input-box-sizing, border-box);
-    height: var(--input-height, fit-content);
+    height: 100%;
```
the component will break for existing places. we can pass 100% from nimble itself. no need of this change

RajuSudhar replied:
Have already exposed the existing variable for the container that wraps input and right img.

sandeepsj commented on src/lib/Input/Input.svelte:
```diff
@@ -218,16 +254,31 @@
     -webkit-appearance: none !important; /* For Safari MWeb */
-    box-shadow: var(--input-box-shadow, 0px 1px 8px #2f537733);
-    border: var(--input-border, none);
+    border: none;
```
Why removing these? the component will break for other already exiting usecases

RajuSudhar replied:
Have add the existing exposed properties below in the class `input` that wraps both input and image.

In Updated Line Number : 264

RajuSudhar replied:
Needs to be moved in order to add the support for icon inside the input.

sandeepsj commented on src/lib/InputButton/InputButton.svelte:
```diff
@@ -104,6 +104,8 @@
     --button-width: 100%;
     --input-border: none;
     --input-focus-border: none;
+        --input-margin: none;
```
formatting issues

RajuSudhar replied:
Yeah

RajuSudhar replied:
Done

sandeepsj commented on src/lib/Toast/Toast.svelte:
```diff
@@ -191,22 +191,27 @@
     left: var(--toast-left, 0);
     right: var(--toast-right, 0);
-    background-color: var(--default-background-color, #87ceeb);
+    background-color: var(--toast-default-background-color, #87ceeb);
```
background-color: var(--toast-default-background-color, var(--default-background-color, #87ceeb); );

For backward compatibility

RajuSudhar replied:
Checked in Nimble, Lighthouse and Harbour. Not in use anywhere.

sandeepsj commented on src/lib/Toast/Toast.svelte:
```diff
@@ -238,25 +243,25 @@
   .success {
     color: var(--toast-success-text, #fff);
-    background-color: var(--toast-background-color, #24aa5a);
+    background-color: var(--toast-success-background-color, var(--toast-background-color, #24aa5a));
```
backward compatibility if needed

RajuSudhar replied:
Yeah already added default var(--toast-background-color) for backward compatibility.

sandeepsj commented on src/lib/Toast/Toast.svelte:
```diff
@@ -238,25 +243,25 @@
   .info {
     color: var(--toast-info-text, #fff);
-    background-color: var(--toast-background-color, #87ceeb);
+    background-color: var(--toast-info-background-color, var(--toast-background-color, #87ceeb));
```
backward compatibility if needed

RajuSudhar replied:
Already added

sandeepsj commented on src/lib/Toast/Toast.svelte:
```diff
@@ -238,25 +243,25 @@
   .warn {
     color: var(--toast-warn-text, #fff);
-    background-color: var(--toast-background-color, #f3a42d);
+    background-color: var(--toast-warn-background-color, var(--toast-background-color, #f3a42d));
```
backward compatibility if needed

RajuSudhar replied:
Added.

sandeepsj commented on src/lib/Input/Input.svelte:
```diff
@@ -244,6 +309,58 @@
+
+  .input-image {
+    cursor: var(--input-image-cursor, pointer);
+    height: var(--input-image-height);
```
height and width can have 30px since it has to fit inside the default of input box

RajuSudhar replied:
done

RajuSudhar committed:
feat: add filterPattern and onFocus properties in Input components

 - Added filterPattern to let input value be filtered based on custom pattern if required
 - Added onFocus event in properties if needed to add some on focus event
